internal/controller/operator/factory/vmagent/vmagent.go

```diff
@@ -1268,6 +1276,59 @@ func buildRemoteWrites(cr *vmv1beta1.VMAgent, ac *build.AssetsCache) ([]string,
 		oauth2ClientID.flagSetting += fmt.Sprintf("%s,", oaclientID)
 		oauth2Scopes.flagSetting += fmt.Sprintf("%s,", oascopes)
 
+		var accessKey, secretKey, ec2Endpoint, region, service, roleARN, stsEndpoint string
+		var useSigv4 bool
+		if rws.AWS != nil {
+			if rws.AWS.AccessKey.Secret != nil || rws.AWS.AccessKey.ConfigMap != nil {
+				if secret, err := ac.LoadKeyFromSecretOrConfigMap(cr.Namespace, &rws.AWS.AccessKey); err != nil {
```

Comment:
It doesn't look like a secure way of providing secret values.

Possible options:
1) do not allow to specify access key at all
2) convert it into env variable somehow? 
3) create a feature request for vmagent to support file based paths.

Reply:
removed static aws creds